src/txn/txn_rollback_to_stable.c

```diff
@@ -8,76 +8,10 @@
 
 #include "wt_internal.h"
 
-/*
- * __txn_rollback_to_stable_hs_fixup --
- *     Remove any updates that need to be rolled back from the history store file.
- */
-static int
-__txn_rollback_to_stable_hs_fixup(WT_SESSION_IMPL *session)
-{
-    WT_CONNECTION_IMPL *conn;
-    WT_CURSOR *cursor;
-    WT_DECL_RET;
-    WT_ITEM hs_key, hs_value;
-    WT_TXN_GLOBAL *txn_global;
-    wt_timestamp_t durable_timestamp, hs_timestamp, rollback_timestamp;
-    uint64_t hs_txnid;
-    uint32_t hs_btree_id, session_flags;
-    uint8_t prepare_state, upd_type;
-
-    conn = S2C(session);
-    cursor = NULL;
-    session_flags = 0; /* [-Werror=maybe-uninitialized] */
-
-    /*
-     * Copy the stable timestamp, otherwise we'd need to lock it each time it's accessed. Even
-     * though the stable timestamp isn't supposed to be updated while rolling back, accessing it
-     * without a lock would violate protocol.
-     */
-    txn_global = &conn->txn_global;
-    WT_ORDERED_READ(rollback_timestamp, txn_global->stable_timestamp);
-
-    WT_RET(__wt_hs_cursor(session, &session_flags));
-    cursor = session->hs_cursor;
-
-    /* Discard pages we read as soon as we're done with them. */
-    F_SET(session, WT_SESSION_READ_WONT_NEED);
-
-    /* Walk the file. */
-    while ((ret = cursor->next(cursor)) == 0) {
-        WT_ERR(cursor->get_key(cursor, &hs_btree_id, &hs_key, &hs_timestamp, &hs_txnid));
-
-        /* Check the file ID so we can skip durable tables */
-        if (hs_btree_id >= conn->stable_rollback_maxfile)
-            WT_PANIC_RET(session, EINVAL,
-              "file ID %" PRIu32 " in the history store table larger than max %" PRIu32,
-              hs_btree_id, conn->stable_rollback_maxfile);
-        if (__bit_test(conn->stable_rollback_bitstring, hs_btree_id))
-            continue;
-
-        WT_ERR(cursor->get_value(cursor, &durable_timestamp, &prepare_state, &upd_type, &hs_value));
-
-        /*
-         * Entries with no timestamp will have a timestamp of zero, which will fail the following
-         * check and cause them to never be removed.
-         */
-        if (rollback_timestamp < durable_timestamp) {
-            WT_ERR(cursor->remove(cursor));
-            WT_STAT_CONN_INCR(session, txn_rollback_hs_removed);
-        }
-    }
-    WT_ERR_NOTFOUND_OK(ret);
-err:
-    WT_TRET(__wt_hs_cursor_close(session, session_flags));
-
-    F_CLR(session, WT_SESSION_READ_WONT_NEED);
-
-    return (ret);
-}
-
 /*
  * __txn_abort_newer_update --
- *     Abort updates in an update change with timestamps newer than the rollback timestamp.
+ *     Abort updates in an update change with timestamps newer than the rollback timestamp. Also,
+ *     clear the history store flag for the first stable update in the update also.
```

Comment:
There are two 'also' in the sentence. Remove the 2nd one.

Reply:
Fixed.